packages/shared/src/lib/core/wallet/actions/send/sendTransactionFromEvm.ts

```diff
@@ -15,26 +15,29 @@ import {
 } from '@core/activity'
 import { IAccountState, getAddressFromAccountForNetwork } from '@core/account'
 import { updateL2BalanceWithoutActivity } from '../updateL2BalanceWithoutActivity'
+import { sendSignedEvmTransaction } from '@core/wallet/actions/sendSignedEvmTransaction'
+import { CallbackParameters } from '@auxiliary/wallet-connect/types'
 
 export async function sendTransactionFromEvm(
     preparedTransaction: EvmTransactionData,
     chain: IChain,
-    callback?: () => void
+    signAndSend: boolean,
+    callback?: (params?: CallbackParameters) => void
 ): Promise<void> {
     const account = getSelectedAccount()
-    const provider = chain.getProvider()
 
     await checkActiveProfileAuth(
         async () => {
-            const chainId = chain.getConfiguration().chainId
-            const coinType = chain.getConfiguration().coinType
-            const transactionReceipt = await signAndSendEvmTransaction(
-                preparedTransaction,
-                chainId,
-                coinType,
-                provider,
-                account
-            )
+            const signedTransaction = await signEvmTransaction(preparedTransaction, chain, account)
+
+            if (!signAndSend || !signedTransaction) {
+                if (callback && typeof callback === 'function') {
```

Comment:
i think here and line `54` we dont need the ` && typeof callback === 'function'` part,  as thats already taken care of with TS

Reply:
It's useful for runtime checking, but I checked and we specify the callback each time, so I applied your suggestion.